src/libos/src/process/do_spawn/mod.rs

```diff
@@ -107,6 +108,59 @@ fn new_process(
     spawn_attributes: Option<SpawnAttr>,
     host_stdio_fds: Option<&HostStdioFds>,
     current_ref: &ThreadRef,
+) -> Result<ProcessRef> {
+    let new_process_ref = new_process_common(
+        file_path,
+        argv,
+        envp,
+        file_actions,
+        spawn_attributes,
+        host_stdio_fds,
+        current_ref,
+        None,
+        None,
+    )?;
+    table::add_process(new_process_ref.clone());
+    table::add_thread(new_process_ref.main_thread().unwrap());
+
+    Ok(new_process_ref)
+}
+
+/// Create a new process for execve which will use same parent, pid, tid
+pub fn new_process_for_exec(
+    file_path: &str,
+    argv: &[CString],
+    envp: &[CString],
+    current_ref: &ThreadRef,
+) -> Result<ProcessRef> {
+    let tid = ThreadId {
+        tid: current_ref.process().pid() as u32,
+    };
+    let new_process_ref = new_process_common(
+        file_path,
+        argv,
+        envp,
+        &Vec::new(),
+        None,
+        None,
```

Comment:
Can you please check the stdio status?

Reply:
Based on `do_spawn`, the host_stdio_fd should be set to `None` because it should have been set by `occlum_ecall_new_process` at the Occlum entry.